[READY]Replaces Trekchems with Cobby Category 2 Chems + Chem Tweaks

Krysonism opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR aims to start breaking up the chemical patch meta by providing 3 new chems that provide bonuses for other application methods. 
Another goal is to also provide more interesting competitors to bicardine, kelotane and anti-toxin since maintainers have expressed some interest in making the trekchems unmixable. 

This PR also contains new sprites, medical devices and some minor code changes to support the new medicines. 

Charcoal has also been changed to require ingestion, since injecting or externally applying an insoluble granular substance makes no sense. I have also chosen to reduce the amount of charcoal in each toxin treatment kit, but do not worry, there is still more than enough charcoal round start to last the entire round.

A brief description of the new stuff:

**Trophazole** - a brute healer that when taken by mouth lets your body turn nutriment into healing peptides for some time, makes the chef feel appreciated and allows for reagent efficient self healing.

**Rhigoxane** - a burn healer that helps you cool down, chills and extinguishes a burning patient if applied with a spray bottle.

**Thializid** - A mixable toxin healer with a high healing potential if applied using an IV drip, very risky to use if improperly diluted or if the patient has poor liver health.

**IV bags** - We now have plastic IV bags that can be easily filled with custom chemicals, the bag can then be sealed for spill proofing and tamper resistance. Make those chem mixes!

I have also added empty spray bottles and blood packs to the med lathe and a special suspicious spray bottle in the chemdrobe contraband section. 

The new spray bottles and a couple of the old ones have new sprites. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Medical gameplay is still lacking in many areas and i believe having more tools to choose from as a doctor will make the role more appealing. This is also true for traitorous doctors since all these chems have overdose thresholds unlike the common goon medicines. <!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Trophazole - a new hearty brute medicine.
add:  Rhigoxane -- a new chilly burn medicine.
add: Thializid - a new high-stakes tox medicine.
del: The 3 basic trek chems are now unmixable.
add: IV medicine bags and blood packs are available in the medlathe.
add: suspicious spray bottles have been added to the chemdrobe contraband compartment.
add: The medical lathe can now print spray bottles.
del: Removed the charcoal syringes from the medkits.
balance:  There is now slightly less charcoal available roundstart.
imageadd: Added new spray bottle & iv bag sprites.
code: trans_to can now react the reagents if you want.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

![spraycontrastgit](https://user-images.githubusercontent.com/49783092/59546319-6a390c00-8f2b-11e9-9731-5cad0170ddff.png)
![spraysuspiciousgit](https://user-images.githubusercontent.com/49783092/59546320-6a390c00-8f2b-11e9-94e2-34c80519357c.png)

Krysonism committed:
Merge remote-tracking branch 'upstream/master'

Krysonism committed:
Merge pull request #3 from tgstation/master

update fork

Krysonism committed:
Merge pull request #5 from tgstation/master

update master

Krysonism committed:
Adds adjustLiverLoss & getLiverLoss procs to carbon

Krysonism committed:
Adds trophazole

Adds trophazole, a brute healer that makes doctors take a more holistic approach to heal, It is highly reagent efficient by converting nutriment to healing peptides. Creates an avenue for cooperation between kitchen and  medbay.

Krysonism committed:
Adds rhigoxane

Rhigoxane is a chilly burn healer that works best in a spray bottle and is capable of extinguishing fires.

Krysonism committed:
Adds thializid

This tox healer has great potential and mixability, balanced by requiring the doctor to take note of the patients liver health as well as using proper dilution and an iv pole; that wont fit in a powergamers pocket.

Krysonism committed:
Charcoal  + trans_to changes

Makes charcoal only work when ingested, and adds a new proc in reagent holder, lets trans_to react the transferred reagents if a method is chosen.

Krysonism committed:
Merge pull request #7 from tgstation/master

update fork again

Krysonism committed:
Merge branch 'master' into cocomed

Krysonism committed:
Medical + Syndicate spray bottles

Medical spray bottles have been added to techweb under the biotech node for medical vapor application, alt click to change their skin.
Two suspicious spray bottles have been added as expensive contraband items in the chemdrobe. They have one extra spray range and a random suspicious skin from a pool of 8.
The large spray bottle and the space cleaner bottle also have new icons.
In addition trophazole and MMF have new colors.

Krysonism committed:
Adds chempacks

the IV medicine bag, or chem pack is the blood pack analogue for chems, it is easy to fill like a beaker but can also be permanently sealed to prevent spillage and mitigate tampering. It can be renamed with a pen. Both the new chem pack and the old blood pack have been added to the medical lathe and biotech node.

Krysonism committed:
Adds recipes to thializid and trophazole

Also fixes the rhigoxane recipe

plapatin commented:
thread theme:
https://www.youtube.com/watch?v=GXFSK0ogeg4

wesoda25 commented:
PRs like this and the stasis one are actually making medbay fun again, I’m excited. 

Although one criticism that I feel should be addressed is that with our already over saturated chemical list, do you think it might be a good idea to remove some stuff? While its interesting and fun to have multiple chemicals for the same purpose, its also overwhelming for someone first learning chemistry, and the jobs learning curve just keeps on getting steeper.

Krysonism commented:
> 
> 
> PRs like this and the stasis one are actually making medbay fun again, I’m excited.
> 
> Although one criticism that I feel should be addressed is that with our already over saturated chemical list, do you think it might be a good idea to remove some stuff? While its interesting and fun to have multiple chemicals for the same purpose, its also overwhelming for someone first learning chemistry, and the jobs learning curve just keeps on getting steeper.

I started this PR when it looked like the trekchems would become unmixable. I personally still think that is the way to go, but Cobby will have to decide when it is time.

Krysonism commented:
The IV bags can still be injected with a syringe after they have been sealed, either to reuse the empty bag or for sabotage.
I might look into a way of recycling the empty bags, can we insert items into the department lathes yet like we can with the autolathe?

dr-bee commented:
Not having peptides taste of mint frosting is a waste of a possible reference.

RandolfTheMeh commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1312,5 +1319,150 @@
+	. = 1
+
+/datum/reagent/medicine/thializid/overdose_process(mob/living/carbon/M)
+	M.adjustLiverLoss(2)
```
You can do liver damage by doing M/human.applyLiverDamage(25), though I'm not sure how kosher that is with the coders. Just kinda find it weird to add duplicates of liver procs. Can use this to get liver damage, too.

HomocidlePancakes commented:
You should loved the OD on  Troph and Rhigo to 15, or people will stack the everlasting shit out of heals.

optimumtact commented:
I am a bit worried about having too many medical options, I mean after all we only have like four/five major damage types but i"ll leave that up to cobby

Dorsisdwarf commented:
I like these chems but whats the deal with the worldsalad names?

Krysonism commented:
> 
> 
> I like these chems but whats the deal with the worldsalad names?

**Trophazole** was created by using the greek prefix _tropho-_ meaning relating to nourishment, and the chemical suffox _-azole_ meaning it contains a five membered nitrogenous heterocycle.

**Rhigoxane** - was created using the prefix _rhigo-_. relating to frost and the suffix _-oxane_ which means it contains tetrahydropyran ring, a type of cyclic ether.

**Thializid** - The prefix thia- means it contains sulfur, the zid means at least in my headcannon that it contains an azide group, i choose the germanic version(azid) make it sound more international.

I chose to make up new names rather then using existing real world drugs because mundane drugs having super space healing powers triggers my chem autism.

zombiegums commented:
Very nice. Trophazole is probably my favourite out of the lot.

Barhandar commented:
There should be no difference between spray and patch application - they're both topical. As such in the unlikely events this PR gets merged and I make the slight method refactor, I'll move it to inhalation.

>because mundane drugs having super space healing powers triggers my chem autism.

Renaming goonchems (blood-clotter, topical painkiller, internal painkiller, anabolic steroid, bronchodilator) when.

Mickyan commented:
So now we have two completely different types of similarly named medical sprays, why?

Krysonism commented:
> 
> 
> So now we have two completely different types of similarly named medical sprays, why?

Because the medspray should never have been named that since they are not sprays but refillable multiuse patches.

If i can figure out how to replace the boxes on the maps i can rename the medsprays to medgels.

Tlaltecuhtli commented on code/modules/reagents/reagent_containers/chem_pack.dm:
```diff
@@ -0,0 +1,48 @@
+/obj/item/reagent_containers/chem_pack
```
you can just put a beaker in the iv drop does this do anything important

Krysonism replied:
The chem packs  can be renamed with a pen and are spillproof + harder to sabotage when sealed.
In addition they are less likely to get stolen by someone who needs a beaker. 

Also the doctors intuitively know that it is intended for the IV drip when they see it in the fridge or on a table.  

I think having proper IV bags adds to the medbay atmosphere as well.

Krysonism committed:
Medsprays are now medigels + fixes

chem packs can now be stored in the smartfridge and chem bag.

Medsprays were renamed medigels to reduce confusion. Calling them medical sprays made no sense since they apply chems through the patch method. 

The lore for the new medigels  is that they spray a thick gel onto the skin that dries into a film.

Krysonism commented:
> 
> 
> > So now we have two completely different types of similarly named medical sprays, why?
> 
> Because the medspray should never have been named that since they are not sprays but refillable multiuse patches.
> 
> If i can figure out how to replace the boxes on the maps i can rename the medsprays to medgels.

I have renamed the medsprays to medigels to emphasize their patch nature and reduce confusion.

The idea is that they now spray a thick gel that dries into a film.

imsxz commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -226,13 +226,13 @@
-	desc = "A box full of medical sprayers, with unscrewable caps and precision spray heads."
+/obj/item/storage/box/medigels
+	name = "box of medical gels"
+	desc = "A box full of medical gel applicators, with unscrewable caps and precision spray heads."
```
>spray heads

Krysonism replied:
What do you suggest i call them? The idea is that the gel is sprayed over the wound where it dries into a patch like film.

If i can a good suggestion on how to place every instance of spray in the medigel text i am willing to do so.

Barhandar replied:
I don't see an issue, I've got screen-cleaning gel that's sprayed.

ATH1909 commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -398,6 +398,13 @@
+	if(method == INGEST)
+		return
+	M.reagents.remove_reagent(/datum/reagent/medicine/charcoal/, reac_volume) //We really should not be injecting an insoluble granular material.
+	M.reagents.add_reagent(/datum/reagent/carbon, reac_volume) // It's pores would get clogged with gunk anyway.
```
"its" pores, not "it's"

reeeeeeeee

Krysonism replied:
night coding, i will take care of it.

Barhandar replied:
Both are valid but the apostrophe-less version is better because it can't confused with the shorthand for "it is".

Krysonism replied:
fixed

ATH1909 commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1312,5 +1319,151 @@
+	if(method != INGEST)
+		return
+
+	M.reagents.remove_reagent(/datum/reagent/medicine/trophazole/, reac_volume * 0.05)
```
is there supposed to be a slash at the end of trophazole in this?

Krysonism replied:
probably not

anconfuzedrock commented:
I feel thializid isn't explained well enough is it a low od or what and tropeasshole will have a 0% chance of being used but you could make that 10% if you made it heal all dmg types instead of just one

Krysonism commented:
> 
> 
> I feel thializid isn't explained well enough is it a low od or what and tropeasshole will have a 0% chance of being used but you could make that 1% if you made it heal all dmg types instead of just one

Thializid has an overdose threshold of 6, but if injected most of it turns into the more forgiving metabolite oxalizid if the patients liver is healthy. A 40% solution of thializid will heal 5 tox per cycle and purge 2 toxins but is terribly unsafe, even more so if the patient has taken significant liver damage. A 20u solution is pretty safe however if the patient doesnt have a mad liver. Its main advantage is that it like anti-toxin can be mixed with other chems, infact this is encouraged as the drug must be diluted to be used anyhow.

Krysonism commented:
> 
> 
> Not having peptides taste of mint frosting is a waste of a possible reference.

Good idea, i will probably make a commit with some minor changes people have requested tonight.

kittymaster0 commented:
I looked through the code and it looks like medbay only gets 12 total charcoal pills. 6 in each box @ 2 per map and 0 in the vendor. Is this okay?

Krysonism commented:
> 
> 
> I looked through the code and it looks like medbay only gets 12 total charcoal pills. 6 in each box @ 2 per map and 0 in the vendor. Is this okay?

Even after my change, there are 22 pills and 14 bottles on metastation, that is 640u charcoal, enough to heal 6400 tox damage, more than will ever be needed.

I wish there were only like 150u, but i will save that for its own PR in order to not poison this one.

ATH1909 commented:
> I feel thializid isn't explained well enough is it a low od or what and tropeasshole will have a 0% chance of being used but you could make that 1% if you made it heal all dmg types instead of just one

Don't the peptides heal small amounts of burn damage?

Tlaltecuhtli commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -734,9 +734,18 @@
+	burn_heal = 1
+
 /datum/reagent/consumable/caramel
-	name = "Caramel"
```
what

Krysonism replied:
That line is still there when i view the file.

Tlaltecuhtli commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1312,5 +1319,151 @@
+	overdose_threshold = 20
+
+/datum/reagent/medicine/trophazole/on_mob_life(mob/living/carbon/M)
+	M.adjustBruteLoss(-1.5*REM, 0.) // heals 3 brute & 0,5 burn if taken with food. compared to 2.5 brute from bicard + nutriment
```
thats not 0.5

Krysonism replied:
What do you mean?

Krysonism commented:
> 
> 
> > I feel thializid isn't explained well enough is it a low od or what and tropeasshole will have a 0% chance of being used but you could make that 1% if you made it heal all dmg types instead of just one
> 
> Don't the peptides made by trophazole heal small amounts of burn damage?

Yes, 50% chance to heal 1 burn.

Hatterhat commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1312,5 +1319,151 @@
+	if(L.failing)
+		return
+	conversion_amount = reac_volume	* (min(100 -L.damage, 80) / 100) //the more damaged the liver the worse we metabolize
+	to_chat(M, "conversion amount:[conversion_amount]")
```
are these supposed to be here? probably not

Krysonism replied:
Good catch, i used these for debugging. Will remove tomorrow after work when i push all the other fixes

Krysonism committed:
minor fixes + mint frosting

I have tried to fix the problems people have pointed out. And the peptides now taste like mint frosting.

ExcessiveUseOfCobblestone commented:
this is on my radar anons so ill look at it soon tm

Krysonism commented:
I will fix the new merge conflicts once the review gets going.

ExcessiveUseOfCobblestone commented:
Are these meant to replace something? I don't know if I like the idea of having a variety because they aren't really alternatives since they're stackable.

If you're interested in playing around with healing I'd like to see 3 aspects for each main damage:

1) The No med option which is usually the worst. Example is CPR.
2) The Emergency option which is really good at healing the primary damage but it comes at a health trade. I guess perf is currently the only example of this. Maybe calomel at high health.
3) The "Best" option which is alright at healing, comes at no health cost, but is difficult to make and the least "macro friendly" or otherwise difficult to obtain. Salbu is a good example.

If we're going to add healing options, I would prefer they be 1 or 2. As of right now I don't really see 3's being okayed by me unless some of the other chems be removed.

Personally I don't find the medicines themselves "fun" aspects of the job (well not since foamestry died out but that was because it was OP). What I want from MD is to challenge players to make tough decisions in order to keep players alive to the best of their ability. I want to have to occasionally spin plates after a murderbone, make a quick decision between waiting for a chem to be made vs giving them the potentially lethal emergency chem, stuff like that. That way if the obstacle is overcomed and the players are still kicking, I feel like I defeated a boss of sorts.

I guess I'm trying to give medbay the dark souls appeal??? 

As for the patches, I don't think i'm a fan of the blizzard style of balancing, at least for ss13. I don't think we should "shake the meta" of patches by having a competitive alternative while we still have plenty of alternatives that aren't even bad themselves. They're just not as stupidly strong as patches lol.

Krysonism commented:
> 
> 
> Are these meant to replace something? I don't know if I like the idea of having a variety because they aren't really alternatives since they're stackable.
>

These were originally intended to replace bicard, kelotane and anti-toxin, but i did not want to make those unmixable without checking with you first since you had previously closed such a PR.

I would gladly make a comitt removing their recipes, if you feel it is the way forward.

As for emergency chems, outside calomel, would they not just simply be outclassed by atropine unless made massively powerful? 

Also, i think that medbay being patchworld is hindering you vision of emergency medicine in a away since they magically bypass masks and hardsuits, making triage trivial.

Currently,  2 of the 4 base damage types are healed by chems that can instantly get the patient on their feet and the only situation where are doctors has trouble getting a patient out of crit in a very short time are serious poisoning and heavy bloodloss.

If we want to make triage more difficult, we need more ways for a spaceman to get hurt(organ damage trauma is my favourite suggestion), increase the rate of bloodloss and take a serious look at the existing chems(styptic, silversulfa, synthflesh and atropine).

Stasis beds gives you ample time to think and prepare to treat difficult cases and the defibs gives you a mulligan if you fuck up.

Overall i think medbay needs a serious rework if you are going  achieve your vision.

optimumtact commented:
our commitment is ironclad

ExcessiveUseOfCobblestone commented:
> These were originally intended to replace bicard, kelotane and anti-toxin, but i did not want to make those unmixable without checking with you first since you had previously closed such a PR.

These sound equally powerful though

> As for emergency chems, outside calomel, would they not just simply be outclassed by atropine unless made massively powerful?

anti-crit chems need to be changed. I'd like to remove all chems that heal a plethora of damages unless they're well balanced to be in category 2/3.

> Also, i think that medbay being patchworld is hindering you vision of emergency medicine in a away since they magically bypass masks and hardsuits, making triage trivial.

I agree.

> Currently, 2 of the 4 base damage types are healed by chems that can instantly get the patient on their feet and the only situation where are doctors has trouble getting a patient out of crit in a very short time are serious poisoning and heavy bloodloss.

Yes, i'd like to change that

> If we want to make triage more difficult, we need more ways for a spaceman to get hurt(organ damage trauma is my favourite suggestion), increase the rate of bloodloss and take a serious look at the existing chems(styptic, silversulfa, synthflesh and atropine).

I was playing around with upping the trauma system to apply to all organs and bodyparts.

> Stasis beds gives you ample time to think and prepare to treat difficult cases and the defibs gives you a mulligan if you fuck up.

Stasis beds are an intermediate fix to sleepers which were mega cringe. Cryo is also a pain.

I can lock stasis beds behind techwebs but sleepers completely null out category 1 options.

Defibs are fine for now, I'd honestly like to see cloning locked fully behind techwebs or made to be not worth doing at T1 unless you literally can't do anything else.

> Overall i think medbay needs a serious rework if you are going achieve your vision.

I think we have an alright foundation, we just need to grow it and prune the cringe bits :P

Krysonism commented:
I will remove trekchem mixing and add some unavoidable liver damage to thializid(making it a type 2 chem)  when i get home from work and hopefully this PR will be more palatable to the maintainers.

lordpidey commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -398,6 +398,13 @@
+/datum/reagent/medicine/charcoal/reaction_mob(mob/living/carbon/M, method=TOUCH, reac_volume)
+	if(method == INGEST)
+		return
+	M.reagents.remove_reagent(/datum/reagent/medicine/charcoal/, reac_volume) //We really should not be injecting an insoluble granular material.
```
Question.

If someone takes a 50u charcoal pill, then gets 50u of charcoal splashed on them, won't this empty their bloodstream of charcoal?

Krysonism replied:
Yes it probably will. I think i will move all the reagent conversion stuff to its own on_transfer proc to avoid this kind of stuff.

lordpidey commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1312,5 +1319,148 @@
+		M.adjustToxLoss(-1.5 * min(2, reac_volume) * REM, 0)	  //This is to promote iv pole use for that chemotherapy feel.
+	if(!locate(/obj/item/organ/liver) in M.internal_organs)
+		return
+	var/obj/item/organ/liver/L = M.internal_organs_slot[ORGAN_SLOT_LIVER]
```
It would probably be better to remove the above if statement on L#1408, and replace it with a null check against L.  That way you only iterate through organs once, and incase someone somehow shoves a liver in the wrong slot, it won't break.

lordpidey commented on code/modules/reagents/reagent_containers/chem_pack.dm:
```diff
@@ -0,0 +1,48 @@
+/obj/item/reagent_containers/chem_pack
+	name = "intravenous medicine bag"
+	desc = "A plastic pressure barg, or 'chem pack', for IV administration of drugs. It is fitted with a thermosealing strip."
```
barg?

lordpidey commented on code/modules/reagents/reagent_containers/chem_pack.dm:
```diff
@@ -0,0 +1,48 @@
+		filling.color = mix_color_from_reagents(reagents.reagent_list)
+		add_overlay(filling)
+
+/obj/item/reagent_containers/chem_pack/AltClick(mob/user)
```
Perhaps add a clumsy check, causing you to spill the contents on yourself?

CPTANT commented:
On their own these chemicals sound a bit useless since, styptic powder, sulfur sulfadiazine and charcoal can still be used unlimited without downsides. Although it's nice the burn medication cools you down and puts out fires. 

There is also Salicyclic Acid and Oxandrolone for brute/burn which are in my opinion completely useless since they are  inferior to simply applying more burn/brute patches.

What chemistry really needs is a purge of all the redundant stuff and a general rebalance so stuff that is actually better at healing is also harder to make.

There should be one basic healing chemical for burn/brute/toxin each which is easy to make but also slow acting (so not instantly healing and infinitely stackable like patches) and one superior chemical (with perhaps a "special" effect) for each that is harder to make, requiring either input resources from botany/mining or research gated

Krysonism commented:
> 
> 
> On their own these chemicals sound a bit useless since, styptic powder, sulfur sulfadiazine and charcoal can still be used unlimited without downsides.

As you can see above the feedback i got from Cobby was that these didn't have enough downsides. This probably means that styptic, sulfa and charcoal are not long for this world, in their current shape at least.

ExcessiveUseOfCobblestone commented:
Is there a driver for just removing the recipe's but not the reagents themselves?

Krysonism commented:
> 
> 
> Is there a driver for just removing the recipe's but not the reagents themselves?

I don't know how it would work in some cases, like botany getting trophazole and the remaining sleepers having thializid. I am willing to remove the trekchems completely if you think that is more appropriate.

Krysonism committed:
Merge pull request #9 from tgstation/master

update fork

Krysonism committed:
Merge branch 'master' into cocomed

Fixes the Nemvar induced bucket related merge conflict

Krysonism committed:
Readds my cleaner sprites

ExcessiveUseOfCobblestone commented:
They **should** go but that prob should be separated PR since you have to account for medibots and stuff.

Just removing them from crafting still effectively replaces them in the usecase we care about so i'm okay with that.

Krysonism committed:
Adds liver damage to thializid + adds on_transfer proc

Thializid has a liver cost, the cost is worse if you have already sustained liver damage. Fixes weirdness by moving the reagent conversion effects from reaction_mob to on_transfer

Krysonism committed:
Fixes chem bag examine

The chem bag examine now werks, also fixed a typo in the description.

Krysonism committed:
Adds a clumsy check to sealing the chem pack

Krysonism committed:
Makes kelo, bicard and anti-tox unmixable.

I have retained the tricord recipe for now in case someone manages to collect all 3 through non-dispenser means. Think of it like collecting all the dragon balls but lame.

Krysonism committed:
Makes damp rag apply on_transfer effects

CPTANT commented:
Uh, making kelo,bicard and anti-tox unmixable is a pretty large change from the original intent of this PR.

Krysonism committed:
glass and food now work with trophazole

glasses and food now call the on_transfer proc via trans_to

ExcessiveUseOfCobblestone commented:
> Uh, making kelo,bicard and anti-tox unmixable is a pretty large change from the original intent of this PR.

It's only inscope because I won't merge yet another basic healing chem to an already saturated list of healing chems.

ExcessiveUseOfCobblestone commented:
That said the charcoal changes might need to be separated since it's rationale has been loreposting instead of "breaking the patch meta" or what have you.

optimumtact commented:
@Krysonism @ExcessiveUseOfCobblestone I just wanted to say thanks I really appreciate you both working on this.

CPTANT commented:
Still literally no reason to use chemicals with downsides when Styptic, sulfur and charcoal just heal without them. And styptic and sulfur do so instantly.

Krysonism commented:
That is a better title. thanks Cobby. 

Another reason for the charcoal change was that putting it in the same drip as thializid seemed a bit too convenient . The synergy is still there but now you have to work just slightly more for it.

On a side note: 
There is also something like 890u charcoal at round start on meta  before the modest reduction in this PR. That is enough to heal 8900 toxin damage, more than you will ever need. This is something i plan on addressing in another PR after this one is satisfactory and merged.

Angustmeta commented:
Thializid:
`required_reagents = list(/datum/reagent/sulfur = 1, /datum/reagent/fluorine = 1, /datum/reagent/toxin = 1, /datum/reagent/nitrous_oxide = 2)`

As others have expressed, this chem is much worse than charcoal and much harder to make and use. Toxin as a reagent is very limited supply, and nitrous oxide is harder to make than even charcoal itself imo, and can explode. 

The rhigoxane recipe looks fun though. I like the idea of a spray bottle chem but vapor tends to suffer from permeability protection blockage. I noticed some related changes in the chem holder file that I don't know what they do so maybe that's addressed akready. 

The new spray bottle sprites look syndicate and out of place. Not an improvement. I assume the janitor will now start with a syndicate space cleaner. 

I consider these to be interesting niche/unique chems as they are now, and not base medicines.

Krysonism commented:
> 
> 
> Thializid:
> `required_reagents = list(/datum/reagent/sulfur = 1, /datum/reagent/fluorine = 1, /datum/reagent/toxin = 1, /datum/reagent/nitrous_oxide = 2)`
> 
> As others have expressed, this chem is much worse than charcoal and much harder to make and use. Toxin as a reagent is very limited supply, and nitrous oxide is harder to make than even charcoal itself imo, and can explode.
> 
> The rhigoxane recipe looks fun though. I like the idea of a spray bottle chem but vapor tends to suffer from permeability protection blockage. I noticed some related changes in the chem holder file that I don't know what they do so maybe that's addressed already.
> 
> The new spray bottle sprites look syndicate and out of place. Not an improvement. I assume the janitor will now start with a syndicate space cleaner.
> 
> I consider these to be interesting niche/unique chems as they are now, and not base medicines.

thializid has the potential to heal 5 tox and selectively purge 2 toxin per cycle versus charcoal which heals two and purges 1 of everything.

Charcoal needs to be taken by mouth, is safe and convenient, but thializid needs to be injected using and IV drip for optimal results and is very dangerous when misused. 

I assume charcoal will be the most common chem for treating tox, but experienced doctors can now make great high performance healing mixes.

These are very different chems

As for the spray cleaner sprites. The black and red ones are all skins for the suspicious spray bottle from the chemdrobe contraband section. 

The upper row of sprites are the ones that belong to the regular cleaner and  medical spray bottles.

In my testing, rhigoxane seemed to work well enough on people with regular crewmember clothing, of course it wont work on some people who wear hardsuits and such but that is the price you pay for instant application.

Tlaltecuhtli commented:
why would i use that when pentetic is an 1 click macro tho which has more functionalities like purging rads

Krysonism commented:
> 
> 
> why would i use that when pentetic is an 1 click macro tho which has more functionalities like purging rads

Faster healing, mixable with other chems such as atropine, pefluoro, salbutamol, brute and burn healing chems.

Angustmeta commented:
Fair points. I misunderstood and thought the top row was the old sprites and the red ones were the new. 

Looks good.

optimumtact commented:
@ExcessiveUseOfCobblestone @Krysonism is this ready?

ExcessiveUseOfCobblestone commented:
I'm fine with it concept-wise but need to review

Krysonism commented:
This is an ambitious project for my rudimentary coding ability so i suspect another round of fixes might be in order after review. I think I've fixed everything i that has been pointed out to me and the issues i have noticed in testing.

I am open to suggestions on how to make rhigoxane a more of a type 2 medicine without making it obsolete. I strongly believe styptic and silver sulfa need to be changed if we want less convenient brute and burns healers to see even marginal use.

Trophazole, being base 25% weaker and having lower OD threshold than the seldom used bicard i feel is fairly inoffensive despite not having any drawbacks or damage conversion.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1314,5 +1321,148 @@
+	var/obj/item/organ/liver/L = C.internal_organs_slot[ORGAN_SLOT_LIVER]
+	if(L.failing || !L)
+		return
+	conversion_amount = trans_volume * (min(100 -L.damage, 80) / 100) //the more damaged the liver the worse we metabolize.
```
why are you not using the proc you made here?

Krysonism replied:
I am now and it.. werks.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/recipes/medicine.dm:
```diff
@@ -264,3 +246,24 @@
+	results = list(/datum/reagent/medicine/rhigoxane/ = 5)
+	required_reagents = list(/datum/reagent/cryostylane = 3, /datum/reagent/bromine = 1, /datum/reagent/lye = 1)
+	required_temp = 200
+	is_cold_recipe = 1
```
boo?

Krysonism replied:
Its now TRUE.

ExcessiveUseOfCobblestone commented on code/modules/reagents/reagent_containers/chem_pack.dm:
```diff
@@ -0,0 +1,53 @@
+	if(v > 0)
+		var/mutable_appearance/filling = mutable_appearance('icons/obj/reagentfillings.dmi', "chempack1")
+		filling.icon_state = "chempack[v]"
+		filling.color = mix_color_from_reagents(reagents.reagent_list)
```
usually you just use the master reagent so you can sneak naughty chems in.

Krysonism replied:
The idea was that a big brain doctor might be able to guess what certain packs might contain based on the colors. A too strong thializid mix might appear too neon green.

ExcessiveUseOfCobblestone replied:
I'm not moved either way but I think just seeing an odd color is going to flag knowledgeable people and ruin sabotage potential.

ExcessiveUseOfCobblestone commented on code/modules/reagents/reagent_containers/chem_pack.dm:
```diff
@@ -0,0 +1,53 @@
+		seal()
+		to_chat(user, "<span class='notice'>You seal the bag.</span>")
+
+/obj/item/reagent_containers/chem_pack/proc/seal()
```
why is this a proc?

Krysonism replied:
I guess if someone wants to make a machine or something that autofills and seals iv bags? It felt right when i was writing it.

ExcessiveUseOfCobblestone replied:
if there's no need for it atm it's just creating overhead fnr

Krysonism replied:
I will fix this today

ExcessiveUseOfCobblestone commented on code/modules/reagents/reagent_containers/spray.dm:
```diff
@@ -349,3 +350,49 @@
+			if("sprayer_med_blue")
+				item_state = "sprayer_med_blue"
+				to_chat(user, "item state set.'")
+				return
```
are these debug messages?

reskin_obj should handle setting the item_state too imo.

Krysonism replied:
Yes, good catch

Krysonism replied:
I have removed the debug printout, not sure how to best tackle the reskin_obj thing. I agree that allowing reskin_obj handle item state as well makes sense but i am not sure i am the right man for the job.

 I might be able to implement it competently with some help with the good people of discord.

ExcessiveUseOfCobblestone replied:
i forgot what item_state does but you can just add an arg to reskin_obj prob.

ExcessiveUseOfCobblestone commented on code/modules/reagents/reagent_containers/glass.dm:
```diff
@@ -47,9 +47,7 @@
-			var/fraction = min(5/reagents.total_volume, 1)
-			reagents.reaction(M, INGEST, fraction)
-			addtimer(CALLBACK(reagents, /datum/reagents.proc/trans_to, M, 5), 5)
+			addtimer(CALLBACK(reagents, /datum/reagents.proc/trans_to, M, 5, 1, 1, 0, user, 0, INGEST), 5)
```
T/F defines smh

Krysonism replied:
Fixed

Krysonism committed:
Another round of fixes

Replaces 0/1 values with TRUE/FALSE where appropriate. Removes the debug printout from the spray bottle reskin. Replaces one instance of L.damage with C.GetLiverLoss().

Krysonism committed:
Rhigoxane recipe is now COOLER

200k was too warm since cryostylane cools down so much by itself upon mixing. 
Now the chemist has to choose between adding oxygen for instant cooling or using the heater/cooler to conserve energy.

shellspeed1 commented:
Two things. First, why, does Trophazole use the bodies nutrition, or do you need to take the drug than have nutrition in your system to work? Second, Thializid seems to be more aimed at traitors considering to create it you need toxin which is available from botany(which means you never get it) or an emagged chem dispenser. This is in addition to it causing liver damage which can kill someone.

SgtxRootbeer commented:
>trekchem removal... thanks for that near-silent change

optimumtact commented:
![](https://file.house/MU-A.png)

SgtxRootbeer commented:
effectively a removal, since we know damn well botany isn't gonna grow it.

optimumtact commented:
yes I just wanted to make sure you saw it

shellspeed1 commented:
Can we just merge chem and botany together. Realistically speaking departmental interaction just doesn't occur and this would fix several of the issues we have such as the need for a chem dispenser in botany and the boredom that is chemistry.

Krysonism commented:
> 
> 
> Two things. First, why, does Trophazole use the bodies nutrition, or do you need to take the drug than have nutrition in your system to work? Second, Thializid seems to be more aimed at traitors considering to create it you need toxin which is available from botany(which means you never get it) or an emagged chem dispenser.

Trophazole stimulates the body to release an enzyme that is called mitogen metabolism factor, that enzyme splits protein into healing peptides.

As for thializid, there are like 180u toxin available round start in the nanomed. I doubt it would run out. I like using toxin as an ingredient because it is potentially limited, if more recipes come to use it in the future, the chemists might sometimes have to make a choice which chems to make.

Krysonism commented:
> 
> 
> effectively a removal, since we know damn well botany isn't gonna grow it.

Metastation has 7x 50 potency poppies round start, just grind them if you want bicard.  As for the botanists, i find that you can often get them to grow one thing for you if you engage and talk with them. Ask them for green grapes if you really love kelotane.

ExcessiveUseOfCobblestone reviewed (changes requested):
Formalizing since this been on tracker for a while.

See above for actual review

Krysonism commented:
I will remove the seal proc and fold it into alt-click tomorrow. I still have no idea how to do the reskin_obj thing.

Krysonism committed:
Removes the seal() proc and moves its guts into AltClick

ExcessiveUseOfCobblestone commented:
you can resolve stuff if you actually fixed it btw, doesn't bother me.

ExcessiveUseOfCobblestone commented:
it's obj bound so just move that stuff there

```
/obj/item/reagent_containers/chem_pack/reskin_obj(mob/M)
..()
switch(icon_state)
    if(A)
        item_state = bloo
    blah blah blah
```

Krysonism commented:
> 
> 
> it's obj bound so just move that stuff there
> 
> ```
> /obj/item/reagent_containers/chem_pack/reskin_obj(mob/M)
> ..()
> switch(icon_state)
>     if(bloo)
>         item_state = bloo
>     blah blah blah
> ```

Great, i will put the stuff there when i wake up and after that i think i have addressed everything.

Krysonism committed:
Item_state switching moved to reskin_obj

Krysonism commented:
I feel like this is probably ready. 

Once it is time for the big rebalancing of styptic, sulfa and the like and if we feel that Rhigoxane is too free we could increase the initial cooling factor so it causes more initial frost burns making it unsuitable for freezing or deep crit burn patients.

ExcessiveUseOfCobblestone commented:
did you test that reskin, I just guessed tbh.

Krysonism committed:
Fixes missing medical sprayer left inhand

the sprayer_med_yellow inhand sprite now has the correct name.

Krysonism commented:
> 
> 
> did you test that reskin, I just guessed tbh. (Should work tho but my code is without warranty)

I tested it and it seems to work. The only issues is that the inhand doesn't update until you put the bottle away or put it in your other hand.

I noticed a missing inhand sprite when testing, that has now been fixed.

ExcessiveUseOfCobblestone commented:
oh just call updateicon at the end of that switch 8)

Krysonism committed:
Fixes inhand updating

Krysonism commented:
> 
> 
> oh just call updateicon at the end of that switch 8)

update_icon() did nothing, thankfully am i such a computer wiz and rain man like figure i managed to deduce that M.update_inv_hands() does the trick. This is now [READY] for real.

Yttriums commented:
As a botanist I'm not sure why I would grow trekchems when Earthsblood is still around, but I'm also not good at medical so maybe someone can explain.

Tlaltecuhtli commented:
add them to the earthblood + mannitol plant for like 50% more heals?

Yttriums commented:
But then you risk overdose, whereas I can mainline earthsblood and mannitol forever without problem.

shellspeed1 commented:
Trek chems are pretty easy to get through botany as you can get 2 of em with just ambrosia vulgris. No plants have antitoxin though.

Krysonism commented:
> 
> 
> But then you risk overdose, whereas I can mainline earthsblood and mannitol forever without problem. I like the idea of growing stuff for chem and having more botany-only reagants, but if we want these botany/chem interactions to work out nicely earthsblood might need a healing nerf.
> 
> The mannitol recipe could also change so that it's base reagents aren't available from the seed vendor at roundstart (or grass could lose it's hydrogen gene).

Metastation has plenty of 50 potency poppies round start, so bicard is significantly easier to obtain.

The bicard in poppies should probably be replaced with morphine once pain mechanics are introduced. But replacing the trekchems in plants is another project since it would involve changing quite a few plants and all the balance considering that comes with that stuff.

Earthsblood + mannitol being so incredibly strong is another issue that is hard solve since any nerf would be incredibly controversial. That being said, if botany were unable to make mannitol it would probably be fine.

Yttriums commented:
As a chief abuser of eb+mani I think the nerf would be mostly uncontroversial as I'm getting a little tired of the play pattern.

ExcessiveUseOfCobblestone reviewed (commented):
uh this is a review from last week so if it is dumb just lol it

Krysonism committed:
Update code/modules/mob/living/carbon/damage_procs.dm

Co-Authored-By: moo <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

Krysonism committed:
Update code/modules/mob/living/carbon/damage_procs.dm

Co-Authored-By: moo <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

Krysonism commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -408,7 +408,7 @@
 	if(!O.reagents || !O.reagents.reagent_list.len) // other empty containers not accepted
 		return FALSE
-	if(istype(O, /obj/item/reagent_containers/syringe) || istype(O, /obj/item/reagent_containers/glass/bottle) || istype(O, /obj/item/reagent_containers/glass/beaker) || istype(O, /obj/item/reagent_containers/spray) || istype(O, /obj/item/reagent_containers/medspray))
+	if(istype(O, /obj/item/reagent_containers/syringe) || istype(O, /obj/item/reagent_containers/glass/bottle) || istype(O, /obj/item/reagent_containers/glass/beaker) || istype(O, /obj/item/reagent_containers/spray) || istype(O, /obj/item/reagent_containers/medigel) || istype(O, /obj/item/reagent_containers/chem_pack))
```
Let me figure out how to do this. It will be done in a couple hours i think.

Krysonism replied:
I also see that i accidentally put a C instead of an O in the last istype

Krysonism replied:
I have added and tested the typecache. It werks.

Krysonism committed:
Tries to add typecache

ExcessiveUseOfCobblestone commented on code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:
```diff
@@ -393,6 +393,19 @@
 /obj/machinery/smartfridge/chemistry
 	name = "smart chemical storage"
 	desc = "A refrigerated storage unit for medicine storage."
+	var/list/chemfridge_typecache
```
can we just put this as a static list in accept_check?

Krysonism replied:
Anything for you Cobby

Krysonism committed:
Moves the typecache and makes it static

optimumtact commented:
@ExcessiveUseOfCobblestone shit or get off the pot mannnnnnnnnn

ExcessiveUseOfCobblestone commented:
his last commit was just 48 hours ago during US holiday so please be gentle
